src/storage/store/node_table.cpp

```diff
@@ -148,7 +153,9 @@ void NodeTable::insert(Transaction* transaction, TableInsertState& insertState)
     }
     const auto localTable = transaction->getLocalStorage()->getLocalTable(tableID,
         LocalStorage::NotExistAction::CREATE);
-    localTable->insert(transaction, insertState);
+    ku_dynamic_cast<LocalTable*, LocalNodeTable*>(localTable)
+        ->insert(transaction, insertState,
+            [&](offset_t offset) { return isVisible(transaction, offset); });
```

Comment:
I think the visibility check here needs to be changed, unless I'm missing something here. For inserting into local node table, the `isVisible` check should be performed on local node table, while it seems to be checking committed part of the node table here through `NodeTable::isVisible()`. We should let LocalNodeTable have its own `isVisible` implementation and not pass in the param?

Reply:
Okay. I'm suspicious that this case is not being covered by the tests as I'd initially added it using a dummy transaction and the number of test failures was unchanged.

Reply:
Yeah, it could be not covered yet. I will add tests on this.